New Entry and Terms General: Programming Paradigms (#1037)
Codecademy/docs: content/general/concepts/programming-paradigms/programming-paradigms.md

---
Title: 'Programming Paradigms'
Description: 'An approach to programming. Depending on the language, the preferred paradigm may differ.'
Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Dynamic Programming'
  - 'Functional Programming'

>   - 'Object Oriented Programming'

Reviewer: We already have an OOP tag. Adding the tag spelled out is going to affect a lot of entries that already have the OOP tag.

>   - 'Object Oriented Programming'

Author: ```suggestion
  - 'OOP'
```

CatalogContent:
  - 'paths/computer-science'
  - 'paths/front-end-engineer-career-path'
---

> A **programming paradigm** is an approach to programming. Depending on the language, the preferred paradigm may differ.

Reviewer 2: ```suggestion
A **programming paradigm** is an approach to programming. Depending on the language, the preferred paradigm may differ. Some languages, such as [C++](https://www.codecademy.com/resources/docs/cpp) and [JavaScript](https://www.codecademy.com/resources/docs/javascript), cover multiple paradigms.
```

> Moving from low-level languages (e.g. assembly language) to high-level languages (e.g. Python), programming paradigms change.

Author: ```suggestion
Moving from low-level languages (e.g. [machine code](https://www.codecademy.com/resources/docs/general/machine-code)) to high-level languages (e.g. [Python](https://www.codecademy.com/resources/docs/python)), programming paradigms change.
```

> Moving from low-level languages (e.g. [machine code](https://www.codecademy.com/resources/docs/general/machine-code)) to high-level languages (e.g. [Python](https://www.codecademy.com/resources/docs/python)), programming paradigms change.

Reviewer 2: ```suggestion
Moving from low-level languages (e.g., [machine code](https://www.codecademy.com/resources/docs/general/machine-code)) to high-level languages (e.g., [Python](https://www.codecademy.com/resources/docs/python)), programming paradigms change.
```
